com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDTemporalFilter.RenderGraph.cs

```diff
@@ -0,0 +1,66 @@
+using UnityEngine.Experimental.Rendering;
+using UnityEngine.Experimental.Rendering.RenderGraphModule;
+using UnityEngine.Rendering;
+
+namespace UnityEngine.Rendering.HighDefinition
+{
+    partial class HDTemporalFilter
+    {
+        class TemporalFilterPassData
+        {
+            public TemporalFilterParameters parameters;
+            public TextureHandle depthStencilBuffer;
+            public TextureHandle normalBuffer;
+            public TextureHandle motionVectorBuffer;
+            public TextureHandle velocityBuffer;
+            public TextureHandle historyDepthTexture;
+            public TextureHandle historyNormalTexture;
+            public TextureHandle noisyBuffer;
+            public TextureHandle validationBuffer;
+            public TextureHandle historyBuffer;
+            public TextureHandle outputBuffer;
+        }
+
+        public TextureHandle Denoise(RenderGraph renderGraph, HDCamera hdCamera, TemporalFilterParameters tfParameters, TextureHandle noisyBuffer, TextureHandle historyBuffer, TextureHandle depthPyramid, TextureHandle normalBuffer, TextureHandle motionVectorBuffer)
+        {
+            using (var builder = renderGraph.AddRenderPass<TemporalFilterPassData>("TemporalDenoiser", out var passData, ProfilingSampler.Get(HDProfileId.TemporalFilter)))
+            {
+                // Cannot run in async
+                builder.EnableAsyncCompute(false);
+
+                // Fetch all the resources
+                passData.parameters = tfParameters;
+                passData.depthStencilBuffer = builder.ReadTexture(depthPyramid);
+                passData.normalBuffer = builder.ReadTexture(normalBuffer);
+                passData.motionVectorBuffer = builder.ReadTexture(motionVectorBuffer);
+                passData.velocityBuffer = renderGraph.defaultResources.blackTextureXR;
```

Comment:
What's the difference between motionVector and Velocity buffers?
Do we always pass a black texture? In this case no need to put it in the pass data but we can access the default resources directly inside the pass code.

Reply:
Velocity is the velocity of the intersected surface by the rays, motion vector and the SS velocity of the pixel that casted the ray. It is always black  in this function, but the shader code is shared